code/modules/mob/living/simple_animal/hostile/retaliate/nymph.dm

```diff
@@ -0,0 +1,373 @@
+			else
+				toggle_ai(AI_ON)
+				user.assimilating = FALSE
+				return
+		else
+			user.melee_damage = 0
+	. = ..()
+
+/mob/living/simple_animal/hostile/retaliate/nymph/spawn_gibs()
+	new /obj/effect/decal/cleanable/insectguts(drop_location())
+	playsound(drop_location(), 'sound/effects/blobattack.ogg', 60, TRUE)
+
+/mob/living/simple_animal/hostile/retaliate/nymph/attack_ghost(mob/dead/observer/user)
+	if(client || key || ckey)
+		to_chat(user, "<span class='warning'>\The [src] already has a player.")
+		return
+	if(!is_ghost_spawn || stat == DEAD)
+		to_chat(user, "<span class='warning'>\The [src] is not possessable!")
+		return
+	var/control_ask = alert("Do you wish to take control of \the [src]", "Chirp Time?", "Yes", "No")
+	if(control_ask == "No" || !src || QDELETED(src) || QDELETED(user))
+		return FALSE
+	if(QDELETED(src) || QDELETED(user) || !user.client)
+		return
+	var/mob/living/simple_animal/hostile/retaliate/nymph/newnymph = src
+	newnymph.key = user.key
+	newnymph.unique_name = TRUE
+	to_chat(newnymph, "<span class='boldwarning'>Remember that you have forgotten all of your past lives and are a new person!</span>")
+
+/mob/living/simple_animal/hostile/retaliate/nymph/proc/update_progression()
+	if(amount_grown < max_grown)
+		amount_grown++
+	if(amount_grown > max_grown)
+		amount_grown = max_grown
+	if(!grown_message_sent && amount_grown == max_grown)
+		to_chat(src, "<span class='userdanger'>You feel like you're ready to grow up by yourself!</span>")
+		grown_message_sent = TRUE
+
+/mob/living/simple_animal/hostile/retaliate/nymph/proc/on_entered(datum/source, atom/movable/arrived, atom/old_loc, list/atom/old_locs)
+	if(isdiona(arrived))
+		if(mind != null || stat == DEAD || is_drone) //Does the nymph on the ground have a mind, dead or a drone?
+			return // If so, ignore the diona
+		var/mob/living/carbon/human/H = arrived
+		var/list/limbs_to_heal = H.get_missing_limbs()
+		if(!LAZYLEN(limbs_to_heal))
+			return
+		playsound(H, 'sound/creatures/venus_trap_hit.ogg', 25, 1)
+		var/obj/item/bodypart/healed_limb = pick(limbs_to_heal)
+		H.regenerate_limb(healed_limb)
+		for(var/obj/item/bodypart/body_part in H.bodyparts)
+			if(body_part.body_zone == healed_limb)
+				body_part.brute_dam = brute_damage
+				body_part.burn_dam = fire_damage
+		balloon_alert(H, "[H] assimilates [src]")
+		QDEL_NULL(src)
+
+/mob/living/simple_animal/hostile/retaliate/nymph/handle_mutations_and_radiation()
+	if(radiation > 50)
+		heal_overall_damage(1,1, 0, BODYTYPE_ORGANIC)
+	. = ..()
+
+/mob/living/simple_animal/hostile/retaliate/nymph/proc/evolve(var/mob/living/simple_animal/hostile/retaliate/nymph/nymphs)
+	if(istype(loc, /obj/item/clothing/head/mob_holder))
+		var/obj/item/clothing/head/mob_holder/L = loc
+		src.loc = L.loc
+		qdel(L)
+
+	src.visible_message(
+		("<span class='warning'>[src] begins to shift and quiver, and after engulfing another nymph, erupts in a shower of shed bark as it splits into a tangle of nearly a dozen new dionaea."),
```

Comment:
"Dionae" is, going by what you told me earlier, already the plural form. The singular being Diona.

Seems like you are trying pluralize the plural?

Reply:
Also you are seemingly missing the closing </span> for these